File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n = numbers.length; 
        int[] solution = new int[2]; 

        for(int i = 0; i < n; i++){
            int difference = target - numbers[i]; 
            for(int j = n - 1; j >= 1; j--){
                if(difference == numbers[j]){
                    solution[0] = i + 1; 
                    solution[1] = j + 1; 
                    return solution; 
                }
            }
        }


        return solution; 
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        #brute force: go through every possible transaction

        #Solution:
        # go through and find the numbers that can make some profit
        # there is some price in index > current, that is more expensive

        profit = set()
        for i in range(len(prices)):
            for j in range(i, len(prices)):
                sale = prices[j] - prices[i]
                if (sale > 0):
                    profit.add(sale)
        
        if (len(profit) > 0): 
            return max(profit)
        else:
            return 0

